components/feature/script/src/main/java/mozilla/components/feature/script/SlowLoadingFeature.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.script
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.script.SlowScriptCallback
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Feature implementation that provides a notification when the
+ * slowScript callback is called and enables the user to
+ * continue or stop running the script.
+ *
+ * @property appContext a reference to the context.
+ * @property sessionId id of the session running the slow script.
+ * @property store reference to the application's [BrowserStore].
+ *
+ * This is where we'd observe state.content.slowLoadingScriptCallback.
+ * When it's null -> everything is okay, nothing to do
+ * When is not null -> post a notification (what is that?) for user to choose next actions:
+ *   - slowLoadingScriptCallback.allowSlowLoadingScript()
+ *   - slowLoadingScriptCallback.stopSlowLoadingScript()
+ */
+class SlowScriptFeature(
+    private val appContext: Context,
```

Comment:
Assuming this is a WIP and there is some reason for the Context in a future revision.